### PR TITLE
store autobalance/autoscaling in Helix

### PR DESCRIPTION
**Pull request by Technoboy-**

### CODE REVIEW

File: uReplicator-Manager/src/main/java/com/uber/stream/kafka/mirrormaker/manager/core/ControllerHelixManager.java [hunk 1]
```diff
@@ -168,6 +170,9 @@ public synchronized void start() {
     _helixAdmin = _helixManager.getClusterManagmentTool();
     _helixPropertyStore = _helixManager.getHelixPropertyStore();
 
+    HelixUtils.updateClusterConfig(_helixManager, AUTO_BALANCING,
```
Comment (yangy0000) [inline uReplicator-Manager/src/main/java/com/uber/stream/kafka/mirrormaker/manager/core/ControllerHelixManager.java]: shall we use `AUTO_BALANCING` value that stored in helix when exists?

Comment (yangy0000) [reply uReplicator-Manager/src/main/java/com/uber/stream/kafka/mirrormaker/manager/core/ControllerHelixManager.java]: @xhl1988

File: uReplicator-Manager/src/main/java/com/uber/stream/kafka/mirrormaker/manager/core/ControllerHelixManager.java [hunk 2]
```diff
@@ -117,14 +123,10 @@
...
   private ZkClient _zkClient;
 
-  private boolean _enableAutoScaling = true;
-  private boolean _enableRebalance;
-
   public ControllerHelixManager(
       KafkaClusterValidationManager kafkaValidationManager,
       ManagerConf managerConf) {
     _conf = managerConf;
-    _enableRebalance = managerConf.getEnableRebalance();
```
Comment (yangy0000) [inline uReplicator-Manager/src/main/java/com/uber/stream/kafka/mirrormaker/manager/core/ControllerHelixManager.java]: Are we stilling planing on honour the default configuration when the override doesn't exists

Comment (yangy0000) [reply uReplicator-Manager/src/main/java/com/uber/stream/kafka/mirrormaker/manager/core/ControllerHelixManager.java]: @xhl1988

File: uReplicator-Manager/src/main/java/com/uber/stream/kafka/mirrormaker/manager/core/ControllerHelixManager.java [hunk 3]
```diff
@@ -1420,27 +1422,29 @@ private static String getSrc(String pipeline) {
...
 
   public void enableAutoScaling() {
-    _enableAutoScaling = true;
+    HelixUtils.updateClusterConfig(_helixManager, AUTO_SCALING, ENABLE);
   }
 
   public boolean isAutoScalingEnabled() {
-    return _enableAutoScaling;
+    return HelixUtils
+        .isClusterConfigEnabled(_helixManager, AUTO_SCALING, _conf.getEnableRebalance());
```
Comment (yangy0000) [inline uReplicator-Manager/src/main/java/com/uber/stream/kafka/mirrormaker/manager/core/ControllerHelixManager.java]: use `true` in here.

Comment (Technoboy-) [reply uReplicator-Manager/src/main/java/com/uber/stream/kafka/mirrormaker/manager/core/ControllerHelixManager.java]: yes, sorry

File: uReplicator-Common/src/main/java/com/uber/stream/kafka/mirrormaker/common/Constants.java [hunk 4]
```diff
@@ -16,9 +16,16 @@
...
 
 public final class Constants {
+
   private Constants() {
   }
 
   public static final String HELIX_OFFLINE_STATE = "OFFLINE";
   public static final String HELIX_ONLINE_STATE = "ONLINE";
+
+  public static final String ENABLE = "enable";
```
Comment (xhl1988) [inline uReplicator-Common/src/main/java/com/uber/stream/kafka/mirrormaker/common/Constants.java]: can you add some comments to explain what enable and disable is used for?

**Pull request merged**